File: Difficulty: Basic/Sum of Digit is Palindrome or not/sum-of-digit-is-palindrome-or-not.cpp
// User function Template for C++

class Solution {
  public:
    bool isDigitSumPalindrome(int n) {
        // Step 1: Calculate sum of digits
        int digit_sum = 0;
        int temp = n;
        while (temp > 0) {
            digit_sum += temp % 10;
            temp /= 10;
        }
        
        // Step 2: Check if digit_sum is palindrome
        int original = digit_sum;
        int rev = 0;
        while (digit_sum > 0) {
            rev = rev * 10 + (digit_sum % 10);
            digit_sum /= 10;
        }
        
        return original == rev;
    }
};

File: Difficulty: Basic/Sum of Digit is Palindrome or not/sum-of-digit-is-palindrome-or-not.py
class Solution:
    def isDigitSumPalindrome(self, n):
        # Step 1: Find sum of digits
        digit_sum = 0
        while n > 0:
            digit_sum += n % 10
            n //= 10
        
        # Step 2: Reverse the digit_sum mathematically
        original = digit_sum
        reverse_num = 0
        while digit_sum > 0:
            reverse_num = reverse_num * 10 + digit_sum % 10
            digit_sum //= 10
        
        # Step 3: Compare original with reversed
        return original == reverse_num
